src/commands/cmd_txn.cc

```diff
@@ -68,13 +69,16 @@ class CommandExec : public Commander {
       return Status::OK();
     }
 
+    auto storage = svr->storage_;
     // Reply multi length first
     conn->Reply(Redis::MultiLen(conn->GetMultiExecCommands()->size()));
     // Execute multi-exec commands
     conn->SetInExec();
+    storage->BeginTxn();
     conn->ExecuteCommands(conn->GetMultiExecCommands());
+    auto s = storage->CommitTxn();
```

Comment:
Just a naive question. If a exception is thrown in `conn->ExecuteCommands` (I didn't review carefully, so didn't know would it throw). The transaction will never recover until another txn begin and do the thing

Reply:
Kvrocks won't throw any exception itself, but if the underlying components do, then Kvrocks should exit without committing if we didn't catch it. Thats to say, for the current implementation, all write operations are grouped into one WriteBatch, so it will be committed all or none.

Reply:
By the way, do we have a flag to check that `ExecuteCommands` triggers error. If it has 5 commands, and the 4th command error. Here it will still try to commit. Am I right?

Reply:
Yes, that's right. Redis also don't check the execute command status, for example:
```
❯ redis-cli
127.0.0.1:6379> MULTI
OK
127.0.0.1:6379(TX)> SET k1 v1
QUEUED
127.0.0.1:6379(TX)> hset k1 field value
QUEUED
127.0.0.1:6379(TX)> EXEC
1) OK
2) (error) WRONGTYPE Operation against a key holding the wrong kind of value
```